### PR TITLE
Protectli adl ppi

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Review by mkopec — DISMISSED** [review]

File: payloads/external/edk2/Kconfig.dasharo [hunk 1]
```diff
@@ -352,4 +352,15 @@ config EDK2_CREATE_PREINSTALLED_BOOT_OPTIONS
 	  Tell UEFIPaylaod to scan available EFI System Partitions and create
 	  boot options from the bootloader binaries found on the disks.
 
+config EDK2_USE_UEFIVAR_BACKED_TPM_PPI
+	bool "Use UEFI variable-backed TPM PPI"
+	default n
+	depends on !EDK2_DISABLE_TPM
+	help
+	  Tell UEFIPaylaod to use TPM PPI buffer in UEFI variables. Wokraround
+	  fro platform-specific issues causing the RAM to not  be preserved
```
Comment (SergiiDmytruk) [inline payloads/external/edk2/Kconfig.dasharo]: Typos:
```suggestion
	  Tell UEFIPaylaod to use TPM PPI buffer in UEFI variables. Workaround
	  for platform-specific issues causing the RAM to not be preserved
```

File: payloads/external/edk2/Makefile [hunk 2]
```diff
@@ -148,7 +148,11 @@ endif
 #
 # The Dasharo repository has the following additional options:
 #
-
+ifneq ($(CONFIG_EDK2_DISABLE_TPM),y)
+ifeq ($(COFNIG_EDK2_USE_UEFIVAR_BACKED_TPM_PPI),y)
```
Comment (SergiiDmytruk) [inline payloads/external/edk2/Makefile]:
```suggestion
ifeq ($(CONFIG_EDK2_USE_UEFIVAR_BACKED_TPM_PPI),y)
```

**Review by SergiiDmytruk — CHANGES_REQUESTED** [review]
Typos.  The problematic one is in `payloads/external/edk2/Makefile`.

**Comment by miczyg1** [comment]
https://github.com/Dasharo/coreboot/pull/604/commits/17897d437676f51e7f6cf19c9092a712c03fcce1 squashed into
https://github.com/Dasharo/coreboot/pull/604/commits/4831c0d23646891c57bfdbed9b3a140eefa0c2d7 for clean history.

File: payloads/external/edk2/Kconfig.dasharo [hunk 3]
```diff
@@ -352,4 +352,15 @@ config EDK2_CREATE_PREINSTALLED_BOOT_OPTIONS
 	  Tell UEFIPaylaod to scan available EFI System Partitions and create
 	  boot options from the bootloader binaries found on the disks.
 
+config EDK2_USE_UEFIVAR_BACKED_TPM_PPI
+	bool "Use UEFI variable-backed TPM PPI"
+	default n
+	depends on !EDK2_DISABLE_TPM
+	help
+	  Tell UEFIPaylaod to use TPM PPI buffer in UEFI variables. Workaround
```
Comment (mkopec) [inline payloads/external/edk2/Kconfig.dasharo]:
```suggestion
	help
	  Tell UEFIPayload to use TPM PPI buffer in UEFI variables. Workaround
```

Comment (miczyg1) [reply payloads/external/edk2/Kconfig.dasharo]: https://github.com/Dasharo/coreboot/compare/349b75f1cc3ad96b1a8c2638b4e170272282bef0..dac568f4e6b5a9d0cce28b5be53f7d2b0acbac15

**Review by SergiiDmytruk — APPROVED** [review]

**Pull request merged**